### PR TITLE
Makefile.dep: make `stdio_uart` a feature required by `ethos`

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
An alternative proposal to https://github.com/RIOT-OS/RIOT/pull/12724. Without touching any boards, with `stdio_uart` being the deciding factor if the application builds or not.

### Testing procedure
Take any of the applications changed in this PR and try to build them for one of the boards previously blacklisted. It still fails:

```
$ BOARD=ruuvitag make -C tests/gnrc_ipv6_ext
make: Entering directory '/home/mlenders/Repositories/RIOT-OS/RIOT/tests/gnrc_ipv6_ext'
There are unsatisfied feature requirements: stdio_uart
/home/mlenders/Repositories/RIOT-OS/RIOT/tests/gnrc_ipv6_ext/../../Makefile.include:811: *** You can let the build continue on expected errors by setting CONTINUE_ON_EXPECTED_ERRORS=1 to the command line.  Stop.
make: Leaving directory '/home/mlenders/Repositories/RIOT-OS/RIOT/tests/gnrc_ipv6_ext'
```

### Issues/PRs references
Alternative to https://github.com/RIOT-OS/RIOT/pull/12724

### CODE REVIEW

File: Makefile.dep [hunk 1]
```diff
@@ -404,7 +404,7 @@ ifneq (,$(filter newlib,$(USEMODULE)))
     USEMODULE += newlib_syscalls_default
   endif
   ifeq (,$(filter stdio_rtt stdio_cdc_acm,$(USEMODULE)))
-    USEMODULE += stdio_uart
+    FEATURES_PROVIDED += stdio_uart
```
Comment (aabadie) [inline Makefile.dep]: I think this is forbidden.

Comment (miri64) [reply Makefile.dep]: Where and why?

**Comment by miri64** [comment]
BTW, with this proposal:

```
CFLAGS="-DUSB_CONFIG_VID=0x1209 -DUSB_CONFIG_PID=0x0001" BOARD=arduino-zero USEMODULE=stdio_cdc_acm make -C tests/gnrc_ipv6_ext -j16
make: Entering directory '/home/mlenders/Repositories/RIOT-OS/RIOT/tests/gnrc_ipv6_ext'
There are unsatisfied feature requirements: stdio_uart
/home/mlenders/Repositories/RIOT-OS/RIOT/tests/gnrc_ipv6_ext/../../Makefile.include:811: *** You can let the build continue on expected errors by setting CONTINUE_ON_EXPECTED_ERRORS=1 to the command line.  Stop.
make: Leaving directory '/home/mlenders/Repositories/RIOT-OS/RIOT/tests/gnrc_ipv6_ext'
```

With #12724 it passes (and pulls in 2 stdio implementations... not sure how and why this works).

File: Makefile.dep (hunk 1 above)
Comment (aabadie) [reply Makefile.dep]: Features should only be provided in Makefile.features files. Makefile.dep files are here to do the resolution of dependencies given the provided features. If we start mixing the concepts we will end up in more mess.

This PR doesn't work `make info-boards-supported`, so Murdock will fail in any case:
```
make -C tests/gnrc_ipv6_ext_frag info-boards-supported --no-print-directory | tr " " "\n" | grep thingy52
thingy52
```
Normally the result should be empty if the board is excluded.

**Comment by miri64** [comment]
I agree with you. This PR is not much better than #12473, so I'm closing it.

**Pull request closed without merging**